core-reactive/src/main/java/io/micronaut/core/async/publisher/Publishers.java

```diff
@@ -442,27 +442,43 @@ public static boolean isConvertibleToPublisher(Object object) {
     /**
      * Attempts to convert the publisher to the given type.
      *
-     * @param object The object to convert
+     * @param object        The object to convert
      * @param publisherType The publisher type
-     * @param <T> The generic type
+     * @param <T>           The generic type
      * @return The Resulting in publisher
+     * @deprecated replaced by {@link #convertPublisher(ConversionService, Object, Class)}
      */
+    @Deprecated(since = "4")
     public static <T> T convertPublisher(Object object, Class<T> publisherType) {
+        return convertPublisher(ConversionService.SHARED, object, publisherType);
+    }
+
+    /**
+     * Attempts to convert the publisher to the given type.
+     *
+     * @param conversionService The conversion service
+     * @param object            The object to convert
+     * @param publisherType     The publisher type
+     * @param <T>               The generic type
+     * @return The Resulting in publisher
+     * @since 4.0.0
+     */
+    public static <T> T convertPublisher(ConversionService conversionService, Object object, Class<T> publisherType) {
```

Comment:
can you annotate the method arguments and return type with the appropriate nullability annotations?